### DISCUSSION TITLE
Obsolete Window attributes

### DISCUSSION THREAD

**Discussion by TangDave** [AvaloniaUI/Avalonia#16462] [Q&A] [answered]
Why are the following 3 lines added by default when adding a new Avalonia Window in Visual Studio?

        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="400"

1. these 2 links no longer work - what's the point of including them if they don't lead anywhere?
2. does anyone need DesignWidth and DesignHeight ? Why not just use the actual Width and Height - the live preview uses these if DesignWidth and DesignHeight  are missing. It seems inefficient to set both when changing Window height and width.

**Comment by maxkatz6** [comment, marked as answer]
> these 2 links no longer work - what's the point of including them if they don't lead anywhere?

It's not just link. These are [XML namespaces](https://www.w3schools.com/xml/xml_namespaces.asp). They don't have to work in browser, they have to identify unique property sets. I.e. two properties with the same name can be defined in different XML namespaces.

> does anyone need DesignWidth and DesignHeight ? Why not just use the actual Width and Height - the live preview uses these if DesignWidth and DesignHeight are missing. It seems inefficient to set both when changing Window height and width.

Valid point, DesignWidth/DesignHeight are optional and only used for convenience.
But this kind of code is easier to remove if you don't need, than add when you need.

**Reply by TangDave** [reply]
I just find it very odd to use an obsolete URL as a namespace. It could be replaced by "Foo" and it would make as much sense.

**Reply by timunie** [reply]
@TangDave that's by convention in xml. https://www.w3schools.com/XML/xml_namespaces.asp

**Reply by TangDave** [reply]
There are so many possible ways to form namespaces - very odd that they settled on meaningless or broken urls.
